pkg/controller/windowsmachineconfig/status.go

```diff
@@ -0,0 +1,103 @@
+package windowsmachineconfig
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	wmcapi "github.com/openshift/windows-machine-config-operator/pkg/apis/wmc/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	k8sapierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type StatusManager struct {
+	client             client.Client
+	name               types.NamespacedName
+	status             *wmcapi.WindowsMachineConfigStatus
+	reconciling        bool
+	degradationReasons map[string][]string
+}
+
+// NewStatusManager returns a new instance of StatusManager
+func NewStatusManager(client client.Client, name types.NamespacedName) *StatusManager {
+	return &StatusManager{
+		degradationReasons: make(map[string][]string),
+		client:             client,
+		name:               name,
+		status:             &wmcapi.WindowsMachineConfigStatus{},
+	}
+}
+
+// addDegradationReason adds the reason and message to the list of degradation causes
+func (s *StatusManager) addDegradationReason(reason, message string) {
+	messages, present := s.degradationReasons[reason]
+	if present {
+		s.degradationReasons[reason] = append(messages, message)
+	} else {
+		s.degradationReasons[reason] = []string{message}
+	}
+}
+
+// setReconciling updates the CR's status to indicate that the reconciler is active
+func (s *StatusManager) setReconciling() {
+	object := &wmcapi.WindowsMachineConfig{}
+	err := s.client.Get(context.TODO(), s.name, object)
+	if err != nil {
+		if k8sapierrors.IsNotFound(err) {
+			return
+		}
+		log.Error(err, fmt.Sprintf("could not get %v", s.name))
+	}
+	wmcapi.SetWindowsMachineConfigCondition(&object.Status,
+		*wmcapi.NewWindowsMachineConfigCondition(wmcapi.Reconciling, corev1.ConditionTrue, "", ""))
+	if err = s.client.Status().Update(context.TODO(), object); err != nil {
+		log.Error(err, fmt.Sprintf("could not update status to %v", object.Status))
+	} else {
+		log.V(1).Info("updated status", "status", object.Status)
+	}
+}
+
+// updateStatus syncs the current status with the cluster and removes the reconciling status condition
+func (s *StatusManager) updateStatus() {
+	object := &wmcapi.WindowsMachineConfig{}
+	err := s.client.Get(context.TODO(), s.name, object)
```

Comment:
This entire logic getting the object first and then updating kind of makes it hard to read, I think you need to just have update here, by the time you reached here you should have current object